File: src/main/java/com/ject/studytrip/dummy/presentation/dto/response/LoadDummyMissionInfoResponse.java
package com.ject.studytrip.dummy.presentation.dto.response;

import com.ject.studytrip.dummy.application.dto.DummyMissionInfo;
import io.swagger.v3.oas.annotations.media.Schema;

public record LoadDummyMissionInfoResponse(
        @Schema(description = "미션 ID") Long missionId,
        @Schema(description = "미션 이름") String missionName,
        @Schema(description = "미션 완료여부") boolean completed) {
    public static LoadDummyMissionInfoResponse of(DummyMissionInfo info) {
        return new LoadDummyMissionInfoResponse(
                info.missionId(), info.missionName(), info.completed());
    }
}

File: src/main/kotlin/com/ject/studytrip/dummy/presentation/response/dto/LoadDummyMissionInfoResponse.kt
package com.ject.studytrip.dummy.presentation.response.dto

import com.ject.studytrip.dummy.application.dto.DummyMissionInfo
import io.swagger.v3.oas.annotations.media.Schema

data class LoadDummyMissionInfoResponse(
    @field:Schema(description = "미션 이름")
    val missionName: String,
    @field:Schema(description = "미션 완료 여부")
    val completed: Boolean,
) {
    companion object {
        @JvmStatic
        fun of(dummyMissionInfo: DummyMissionInfo): LoadDummyMissionInfoResponse =
            LoadDummyMissionInfoResponse(
                dummyMissionInfo.missionName,
                dummyMissionInfo.completed,
            )
    }
}
